src/test/resources/processor_test_files/2184_CastArithmeticOperand/MultipleBinaryOperatorsInMethodCall.java

```diff
@@ -0,0 +1,8 @@
+/* When the arithmetic operation is inside a method call, and an inner binary operator of the arithmetic operation
+is being analyzed, we don't find it directly in the list of arguments of the method call, because that binary operator
+is part of something more. Not finding it in the list of arguments causes an IndexOutOfBoundsException because
+we have index -1. This test checks that that exception doesn't happen.
+ */
+class MultipleBinaryOperatorsInMethodCall {
+    Date myDate = new Date(seconds * 1000 * 10); // Noncompliant
+}
```

Comment:
I'm pretty sure the test failures are because this file isn't complete, and so neither Spoon nor Sonar can properly resolve all types. Try this:

```suggestion
import java.util.Date;

class MultipleBinaryOperatorsInMethodCall {
    int seconds = 1000;
    Date myDate = new Date(seconds * 1000 * 10); // Noncompliant
}
```

Reply:
Ooops, my bad, I will fix that! Thank you!

Reply:
Super easy to miss, I just happened to fix a bunch of these issues a few days ago and so spotted it immediately :)

Reply:
I would take forever.